src/cpp/simplehttpserver.cpp

```diff
@@ -486,15 +487,15 @@ class SimpleHttpServerPrivate : public QObject
 		return true;
 	}
 
-private slots:
+private:
 	void server_newConnection()
 	{
 		if(local)
 		{
 			QLocalSocket *sock = ((QLocalServer *)server)->nextPendingConnection();
 			SimpleHttpRequest *req = new SimpleHttpRequest(maxHeadersSize, maxBodySize);
 			connect(req->d, &SimpleHttpRequest::Private::ready, this, &SimpleHttpServerPrivate::req_ready);
-			connect(req, &SimpleHttpRequest::finished, this, &SimpleHttpServerPrivate::req_finished);
+			finishedConnection = req->finished.connect(boost::bind(&SimpleHttpServerPrivate::req_finished, this, req));
```

Comment:
We need to keep track of a connection per request here, or else every new request overwrites the one connection.